WTWSMS/common/cultures/00_cultures.txt

```diff
@@ -4644,7 +4645,13 @@ byzantine = {
 		
 		founder_named_dynasties = yes
 
-		modifier = default_culture_modifier
+		allow_looting = yes
+
+		character_modifier = {
+			martial = 4
```

Comment:
It is not unbalanced. Becoming Spartan is a long and difficult process that should reap huge rewards. Like you suggested, you cannot instantly become Spartan once you restore Sparta.